HCore-Storage/Client/Impl/AwsStorageClientImpl.cs

```diff
@@ -0,0 +1,663 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Transfer;
+using HCore.Storage.Exceptions;
+using HCore.Storage.Helpers;
+using HCore.Storage.Models;
+using HCore.Storage.Models.AwsStorage;
+using HCore.Web.Exceptions;
+using Newtonsoft.Json;
+
+namespace HCore.Storage.Client.Impl
+{
+    public class AwsStorageClientImpl : IStorageClient
+    {
+        private const int _defaultPageSize = 1000;
+
+        private readonly IDictionary<string, string> _connectionInfoByKey;
+
+        public AwsStorageClientImpl(string connectionString)
+        {
+            _connectionInfoByKey = AwsHelpers.GetConnectionInfoByKey(connectionString);
+        }
+
+        public async Task DownloadToStreamAsync(string containerName, string fileName, Stream stream)
+        {
+            using var amazonS3 = AwsHelpers.GetAmazonS3(_connectionInfoByKey);
+
+            using var getObjectResponse = await GetObjectResponseAsync(amazonS3, containerName, fileName, required: true).ConfigureAwait(false);
+
+            await getObjectResponse.ResponseStream.CopyToAsync(stream).ConfigureAwait(false);
```

Comment:
How is this implemented? Does it read the file first, and then copy, or is it really streamed (without consuming tons of memory). You can test with a super large file.

Reply:
This does not answer my question.

Reply:
Goes to memory